File: src/01-easy/the-river-ii-/solution.cpp
#include <iostream>
#include <algorithm>
using namespace std;
int ds(long n){int s=0;while(n>0){s+=n%10;n/=10;}return s;}
int main(){
    long r;cin>>r;
    for(long x=max(1L,r-45);x<r;x++)if(x+ds(x)==r){cout<<"YES"<<endl;return 0;}
    cout<<"NO"<<endl;
}

File: src/01-easy/the-river-ii-/solution.php
<?php
function ds($n){$s=0;while($n>0){$s+=$n%10;$n=intdiv($n,10);}return $s;}
$r=intval(fgets(STDIN));
$lo=max(1,$r-45);
for($x=$lo;$x<$r;$x++)if($x+ds($x)==$r){echo "YES\n";exit;}
echo "NO\n";
